toxcore/group.c

```diff
@@ -73,69 +129,47 @@ static int realloc_groupchats(Group_Chats *g_c, uint32_t num)
     return 0;
 }
 
+static void setup_conference(Group_c *g)
+{
+    memset(g, 0, sizeof(Group_c));
+    g->keep_join_index = -1;
+    g->live = true;
+}
 
 /* Create a new empty groupchat connection.
  *
  * return -1 on failure.
  * return groupnumber on success.
  */
-static int create_group_chat(Group_Chats *g_c)
+static aint create_group_chat(Group_Chats *g_c)
 {
-    uint32_t i;
+    aint i;
```

Comment:
`num_chats` is `uint16_t`, so you're comparing a signed int to an unsigned below. It used to be unsigned < unsigned. Please change the type of `i` to `uint16_t` or another unsigned type.